controllers/repo_manager/ingress.go

```diff
@@ -134,9 +137,21 @@ func (r *RepoManagerReconciler) pulpIngressController(ctx context.Context, pulp
 		return ctrl.Result{}, err
 	}
 
+	// Ensure ingress specs are as expected
+	if err := r.reconcileObject(ctx, pulp, expectedIngress, currentIngress, conditionType, log); err != nil {
+		log.Error(err, "Failed to update ingress spec")
+		return ctrl.Result{}, err
+	}
+
+	// Ensure ingress labels and annotations are as expected
+	if err := r.reconcileMetadata(ctx, pulp, expectedIngress, currentIngress, conditionType, log); err != nil {
+		log.Error(err, "Failed to update ingress labels")
+		return ctrl.Result{}, err
+	}
+
```

Comment:
Have you done CR modification tests?
I'm asking this because, if the reconcileObject method is working for ingress, this PR is probably also fixing:
https://github.com/pulp/pulp-operator/issues/672

but, unfortunately, I think that our current implementation is not handling other resources than routes and services:
https://github.com/pulp/pulp-operator/blob/f068a015d4c7ac74044a6b95fbe89099dafba257/controllers/repo_manager/utils.go#L391-L429

Reply:
I didn't, I only tested the status condition

Reply:
Hum... in this case, what about removing the above blocks and working on them (we will probably need to modify the `repo_manager/utils.go` with the reconciliation logic for ingress) in a new PR?

Reply:
Or maybe we can just let it the way it is because both methods will just return nil ...